conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderGenerator.java

```diff
@@ -917,4 +914,68 @@ public Boolean visitUnknown(String unknownType) {
             }
         });
     }
+
+    private CollectionType getCollectionType(Type type) {
+        return type.accept(new DefaultTypeVisitor<>() {
+            @Override
+            public CollectionType visitList(ListType _value) {
+                return _value.getItemType().accept(new DefaultTypeVisitor<>() {
+                    @Override
+                    public CollectionType visitDefault() {
+                        return new CollectionType(
+                                ArrayList.class, options.nonNullCollections() && !isOptionalInnerType(type));
```

Comment:
referencing https://github.com/palantir/conjure-java/pull/2274#issuecomment-2040347409, seems any type with optional in it will need a collection that can hold nulls.

Reply:
I don't think that's true, a `list<optional<string>>` should never include null, otherwise we'd have some `optional.empty` elements and some `null` elements, which wouldn't make a lot of sense!

Reply:
(but we should add tests to keep us honest :-)

Reply:
This is current codegen setter for list<any> and nonNullCollections
```
@JsonSetter(value = "items", nulls = Nulls.SKIP, contentNulls = Nulls.FAIL)
public Builder items(@Nonnull Iterable<?> items) {
    checkNotBuilt();
    this.items = ConjureCollections.newArrayList(Preconditions.checkNotNull(items, "items cannot be null"));
    return this;
}
```
From the annotations, jackson won't let us set deser nulls. So I think it is fine if this setter also uses `newNonNullArrayList`

Reply:
👍